File: hivepulse-web/src/domain/statusPage.ts
import type { Tag } from './tag'

export interface StatusPage {
  id: string
  slug: string
  title: string
  description: string
  logo_url: string | null
  accent_color: string
  custom_domain: string | null
  tag_ids: string[]
  created_at: string
}

export interface DailyBucket {
  date: string
  uptime_pct: number
}

export interface PublicMonitorRow {
  id: string
  name: string
  check_type: string
  last_status: 'up' | 'down' | 'unknown'
  uptime_24h: number
  uptime_90d: number
  daily_buckets: DailyBucket[]
}

export interface PublicIncident {
  id: string
  monitor_name: string
  started_at: string
  resolved_at: string | null
  duration_s: number
  error_msg: string
}

export interface PublicStatusPageData {
  title: string
  description: string
  accent_color: string
  logo_url: string | null
  overall_status: 'operational' | 'degraded' | 'outage'
  monitors: PublicMonitorRow[]
  active_incidents: PublicIncident[]
  recent_incidents: PublicIncident[]
}

export interface CreateStatusPageInput {
  title: string
  slug?: string
  description?: string
  logo_url?: string
  accent_color: string
  custom_domain?: string
  tag_ids: string[]
}

export type { Tag }

File: hivepulse-api/internal/domain/status_page.go
package domain

import "time"

type StatusPage struct {
	ID           string    `json:"id"`
	Slug         string    `json:"slug"`
	Title        string    `json:"title"`
	Description  string    `json:"description"`
	LogoURL      string    `json:"logo_url"`
	AccentColor  string    `json:"accent_color"`
	CustomDomain string    `json:"custom_domain"`
	TagIDs       []string  `json:"tag_ids"` // resolved from status_page_tags
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

type PublicMonitorRow struct {
	ID           string        `json:"id"`
	Name         string        `json:"name"`
	CheckType    string        `json:"check_type"`
	LastStatus   string        `json:"last_status"`
	Uptime24h    float64       `json:"uptime_24h"`
	Uptime90d    float64       `json:"uptime_90d"`
	DailyBuckets []DailyBucket `json:"daily_buckets"`
}

type DailyBucket struct {
	Date      string  `json:"date"`
	UptimePct float64 `json:"uptime_pct"`
}

type PublicIncident struct {
	ID          string     `json:"id"`
	MonitorName string     `json:"monitor_name"`
	StartedAt   time.Time  `json:"started_at"`
	ResolvedAt  *time.Time `json:"resolved_at,omitempty"`
	DurationS   int        `json:"duration_s"`
	ErrorMsg    string     `json:"error_msg"`
}

type PublicStatusPageData struct {
	Title           string             `json:"title"`
	Description     string             `json:"description"`
	AccentColor     string             `json:"accent_color"`
	LogoURL         string             `json:"logo_url"`
	OverallStatus   string             `json:"overall_status"`
	Monitors        []PublicMonitorRow  `json:"monitors"`
	ActiveIncidents []PublicIncident    `json:"active_incidents"`
	RecentIncidents []PublicIncident    `json:"recent_incidents"`
}
